control-plane-operator/main.go

```diff
@@ -458,8 +471,18 @@ func NewStartCommand() *cobra.Command {
 				os.Exit(1)
 			}
 
+			var assumeRole *string
+			var localZoneID *string
+
+			if hcp.Spec.Platform.AWS != nil && hcp.Spec.Platform.AWS.SharedVPC != nil {
```

Comment:
why are we introducing this pattern instead of propagating fields to HCP and let the controller reconcile it as in present implementation https://github.com/openshift/hypershift/blob/4ab267a7179127d196b147e40532fb18c4ce65a6/control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go#L147

Reply:
The main reason is that we construct the aws clients for the AWSEndpointServiceReconciler on setup:
https://github.com/openshift/hypershift/blob/4ab267a7179127d196b147e40532fb18c4ce65a6/control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go#L194-L218

For the shared vpc case, the client needs to be setup to assume the shared vpc control plane role if configured.
I'm fine changing that pattern in that reconciler so we don't have to look up the HostedCluster in main.go
tbh, not sure how much overhead is added by creating the client on every reconcile.

Reply:
Follow up tracked with https://issues.redhat.com/browse/HOSTEDCP-2012